### PR TITLE
add a quic.Config option to configure the idle timeout

### LINKED ISSUE #441: Make idle state timeout configurable
I am using QUIC in a non-HTTP setup, and I need to alter the idle state timeout, but I cannot from what is exposed. Can this be made configurable? I hesitate to do it myself as it appears a lot of things just got upheaved in #420.

Also, is there a place for discussion besides these issues?

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #441.

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -371,7 +371,7 @@ func (s *session) idleTimeout() time.Duration {
 	if s.handshakeComplete {
 		return s.connectionParameters.GetIdleConnectionStateLifetime()
 	}
-	return protocol.InitialIdleTimeout
+	return s.config.HandshakeTimeout / 2
```
Comment (lucas-clemente) [inline session.go]: No magic numbers… Either move the 2 to the top of the file (or server_parameters.go) or add a comment.

File: interface.go [hunk 2]
```diff
@@ -100,6 +100,12 @@ type Config struct {
 	// If the timeout is exceeded, the connection is closed.
 	// If this value is zero, the timeout is set to 10 seconds.
 	HandshakeTimeout time.Duration
+	// IdleTimeout is the maximum duration that may pass without any network activity.
```
Comment (lucas-clemente) [inline interface.go]: any *incoming* network activity

File: interface.go [hunk 3]
```diff
@@ -100,6 +100,12 @@ type Config struct {
...
+	// This value only applies after the handshake is completed.
```
Comment (lucas-clemente) [inline interface.go]: Nit: is complete or has completed.

File: session.go (hunk 1 above)
Comment (marten-seemann) [reply session.go]: This also breaks the handshake RTT test, where we test the handshake timeout. I'm not sure what to do about this, it seems like the only way to solve this is making the initial idle timeout configurable as well.

**Comment by marten-seemann** [comment]
I removed the idle timeout before the handshake has completed.
@lucas-clemente: PHAL.

**Pull request merged**